src/MabeEnum/Enum.php

```diff
@@ -54,7 +54,21 @@
         $this->value   = $value;
         $this->ordinal = $ordinal;
     }
+    
+    /**
+     * Returns a list of all enumeration instances of actual implementation 
+     * 
+     * @return Enum[]
+     */
+    public static function getAll()
```

Comment:
I like `getInstances` or `getEnumerators` much more as it's consistent with the name `getConstants` and the term `list` doesn't help and `enum` is ambiguous: enumeration type vs. enumerator element. I'm going to fix up the variable names of `$enum` as well.

I have benchmarked 3 versions by my version is a little bit faster.
Tested an enumeration with 26 elements or PHP-5.5.6:

```php
    final public static function getEnumerators1()
    {
        return array_map('self::getByName', array_keys(self::detectConstants(get_called_class())));
    }

    final public static function getEnumerators2()
    {
        $enumerators = array_keys(self::detectConstants(get_called_class()));
        foreach ($enumerators as & $entry) {
            $entry = self::getByName($entry);
        }
        return $enumerators;
    }

    final public static function getEnumerators3()
    {
        $enums = array();
        foreach (array_keys(self::detectConstants(get_called_class())) as $v) {
            $enums[] = self::getByName($v);
        }
        return $enums;
    }
```

```php
class MyEnum extends \MabeEnum\Enum {
    const A = 'a';
    const B = 'b';
    const C = 'c';
    const D = 'd';
    const E = 'e';
    const F = 'f';
    const G = 'g';
    const H = 'h';
    const I = 'i';
    const J = 'j';
    const K = 'k';
    const L = 'l';
    const M = 'm';
    const N = 'n';
    const O = 'o';
    const P = 'p';
    const Q = 'q';
    const R = 'r';
    const S = 's';
    const T = 't';
    const U = 'u';
    const V = 'v';
    const W = 'w';
    const X = 'x';
    const Y = 'y';
    const Z = 'z';
}

MyEnum::getConstants();

echo 'getEnumerators1(): ';
$start = microtime(true);
for ($i = 0; $i < 10000; $i++) {
    MyEnum::getEnumerators1();
}
$end = microtime(true);
echo ($end - $start) . PHP_EOL;


echo 'getEnumerators2(): ';
$start = microtime(true);
for ($i = 0; $i < 10000; $i++) {
    MyEnum::getEnumerators2();
}
$end = microtime(true);
echo ($end - $start) . PHP_EOL;


echo 'getEnumerators3(): ';
$start = microtime(true);
for ($i = 0; $i < 10000; $i++) {
    MyEnum::getEnumerators3();
}
$end = microtime(true);
echo ($end - $start) . PHP_EOL;
```

```
php bench.php 
getEnumerators1(): 0.53819012641907
getEnumerators2(): 0.60042691230774
getEnumerators3(): 0.60367202758789
```